azerothcore/azerothcore-wotlk
# fix(Core/DB): Brewspewer hardcoded conditions

## Changes Proposed:
- Removed hardcoded conditions from the core
- Implemented creatures with quest gossips and vendors using a White Chat balloon as shown below
- Creature must have a "Speak" Icon name for the upper to work on Creature Template
- Fixed compiler warning due to missing breaks with ``` [[Fallthrough]];```

## Issues Addressed:
- Closes #4650 



## Tests Performed:
- Compiled and tested in Windows 10



## How to Test the Changes:


1. .go c 50145
2. Talk to the NPC and see that there are no gossip menu options
3. The Creature must also have this Icon when hovering:
![image](https://user-images.githubusercontent.com/54484196/111307181-fde3c300-8650-11eb-9099-afc2509bfe89.png)

4. .quest add 3909
5. Talk to the NPC and there are gossip menu options

## Known Issues and TODO List:

- [x] There's a bug that shows his interaction icon to be a vendor bag due to him needing vendor flags for his gossips.

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
